packetery/src/Packetery/Module/Checkout.php

```diff
@@ -271,23 +263,29 @@ public function updateOrderMeta( int $orderId ): void {
 		if ( false === $this->isPacketeryOrder( $chosenMethod ) ) {
 			return;
 		}
+
 		$post = $this->httpRequest->getPost();
-		if ( empty( $post[ Entity::META_CARRIER_ID ] ) ) {
-			$carrierId = $this->getCarrierId( $chosenMethod );
-			if ( $carrierId ) {
-				update_post_meta( $orderId, Entity::META_CARRIER_ID, $carrierId );
-			}
+
+		// Save carrier id for home delivery (we got no id from widget).
+		$carrierId = $this->getCarrierId( $chosenMethod );
+		if ( empty( $post[ Entity::META_CARRIER_ID ] ) && $carrierId ) {
+			update_post_meta( $orderId, Entity::META_CARRIER_ID, $carrierId );
 		}
+
 		if ( $this->isPickupPointOrder() ) {
 			if ( ! wp_verify_nonce( $post['_wpnonce'], self::NONCE_ACTION ) ) {
 				wp_nonce_ays( '' );
 			}
 			foreach ( self::$pickup_point_attrs as $attr ) {
-				if (
-					isset( $post[ $attr['name'] ] ) &&
-					( Entity::META_CARRIER_ID !== $attr['name'] || $post[ $attr['name'] ] )
-				) {
-					update_post_meta( $orderId, $attr['name'], $post[ $attr['name'] ] );
+				$attrName = $attr['name'];
+				if ( ! isset( $post[ $attrName ] ) ) {
+					continue;
+				}
+				$attrValue                  = $post[ $attrName ];
+				$isNotCarrierIdOrIsNotEmpty = Entity::META_CARRIER_ID !== $attrName || $attrValue;
+				$isNotUrlOrIsValidUrl       = Entity::META_POINT_URL !== $attrName || filter_var( $attrValue, FILTER_VALIDATE_URL );
```

Comment:
Pojmenovávej proměnné bez negace. Toto pojmenuj $hasValidPointUrl a uprav výrazy. A to nad tím pojmenuj $hasCarrierId.

Reply:
Tato pojmenování by ale nevyjadřovala potřebnou podmínku. Ve hře byla ještě tato varianta, ale delší a méně přehledná:
$filters = [
	Entity::META_CARRIER_ID => function ( $value ) {
		return ! empty( $value );
	},
	Entity::META_POINT_URL  => function ( $value ) use ( $carrierId ) {
		return ( Repository::INTERNAL_PICKUP_POINTS_ID === $carrierId && filter_var( $value, FILTER_VALIDATE_URL ) );
	},
];
if ( ! isset( $filters[ $attrName ] ) || call_user_func( $filters[ $attrName ], $attrValue ) ) {
	update_post_meta( $orderId, $attrName, $attrValue );
}

Reply:
Co takto?

Reply:
Ano. Myslím si že je takto čitelnější.